src/libraries/Common/tests/System/Net/Sockets/SocketTestExtensions.cs

```diff
@@ -49,5 +49,31 @@ public static (Socket, Socket) CreateConnectedSocketPair()
 
             return (client, server);
         }
+
+        // Tries to connect within the provided timeout interval
+        // Useful to speed up "can not connect" assertions on Windows
+        public static bool TryConnect(this Socket socket, EndPoint remoteEndpoint, int millisecondsTimeout)
+        {
+            IAsyncResult connectResult = socket.BeginConnect(remoteEndpoint, null, null);
+            if (connectResult.AsyncWaitHandle.WaitOne(millisecondsTimeout))
+            {
+                try
+                {
+                    socket.EndConnect(connectResult);
+                    return true;
+                }
+                catch (SocketException)
+                {
+                    return false;
+                }
+                finally
+                {
+                    connectResult.AsyncWaitHandle.Close();
+                }
+            }
+            connectResult.AsyncWaitHandle.Close();
```

Comment:
I'm not understanding the logic here.  It seems like:
1) This is going to leave the connection pending even after TryConnect returns, and 2)
2) It's going to dispose of the wait handle such that when the operation eventually does complete and tries to set this, it's going to blow up.

Can you redo this instead with Socket.ConnectAsync and Socket.CancelConnectAsync?

Reply:
Done! Wasn't aware of `CancelConnectAsync`, thanks.